src/opendlv-statsd.cpp

```diff
@@ -47,27 +59,90 @@ int32_t main(int32_t argc, char **argv) {
         //       would have to be created and registered.
 
         // Handler to receive a GPS message (realized as C++ lambda).
-        auto onGPS = [&dataMutex](cluon::data::Envelope &&env){
+        
+        std::vector<double> lat_queue,lon_queue; //vector is used as a queue
```

Comment:
This is tied to the WGS84 example; how would the solution independent of the known field name look like?

You should think along the lines that messages have identifiers (ie., `cluon::data::Envelope::dataType()` and `cluon::data::Envelope::senderStamp()`), and all fields have unique numerical identifiers.

Please check this example to make a more dynamic solution: https://wandbox.org/permlink/uR2NZwnYZkTwmDac

You can also have a look at the cluonOD42JSON example here that is dynamically processing message specification files: https://github.com/chrberger/libcluon/blob/master/libcluon/tools/cluon-OD4toJSON.hpp